config/default.yml

```diff
@@ -1972,7 +1972,9 @@ Lint/NonAtomicFileOperation:
   Description: Checks for non-atomic file operations.
   Enabled: pending
   VersionAdded: '1.31'
+  VersionChanged: '<<next>>'
   SafeAutoCorrect: false
+  AllowUnsafeRecursiveRemove: true
```

Comment:
What's unsafe about it?

Reply:
We were discussing it here: https://github.com/rubocop/rubocop/issues/10813#issuecomment-1184312234

`rm_r` with check file existence is expected to be rewritten to `rm_rf`. 
While this is certainly an atomic file operation and will not cause the problems that occur with frequent parallel file operations, on the other hand, there is a danger of recursively deleting protected files without warning, which we hope this added parameter will allow the user to adjust. What do you think?

Reply:
Hm... this cop is Lint, not Style. I doubt it makes sense to have such an option in Lint. I'm not sure, but always allowing is one option for `rm_r`.

Reply:
> Hm... this cop is Lint, not Style. I doubt it makes sense to have such an option in Lint.

That is certainly a bit of a concern there. Surely it might be a good idea to always allow it.